File: android/src/main/java/com/relateddigital/flutter/RelatedDigitalStoryViewFactory.java
package com.relateddigital.flutter;

import android.content.Context;
import android.view.View;

import androidx.annotation.NonNull;

import java.util.Map;

import io.flutter.plugin.common.BinaryMessenger;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.StandardMessageCodec;
import io.flutter.plugin.platform.PlatformView;
import io.flutter.plugin.platform.PlatformViewFactory;

public class RelatedDigitalStoryViewFactory extends PlatformViewFactory {
    @NonNull
    private final BinaryMessenger messenger;
    private final MethodChannel channel;

    RelatedDigitalStoryViewFactory(@NonNull BinaryMessenger messenger, MethodChannel channel) {
        super(StandardMessageCodec.INSTANCE);
        this.messenger = messenger;
        this.channel = channel;
    }

    @Override
    public PlatformView create(Context context, int viewId, Object args) {
        final Map<String, Object> creationParams = (Map<String, Object>) args;
        return new RelatedDigitalStoryView(context, viewId, creationParams, channel);
    }
}

File: android/src/main/kotlin/com/relateddigital/flutter/RelatedDigitalStoryViewFactory.kt
package com.relateddigital.flutter

import android.content.Context
import android.view.View
import java.util.Map
import io.flutter.plugin.common.BinaryMessenger
import io.flutter.plugin.common.MethodChannel
import io.flutter.plugin.common.StandardMessageCodec
import io.flutter.plugin.platform.PlatformView
import io.flutter.plugin.platform.PlatformViewFactory

class RelatedDigitalStoryViewFactory internal constructor(
    @NonNull messenger: BinaryMessenger,
    channel: MethodChannel
) : PlatformViewFactory(StandardMessageCodec.INSTANCE) {
    @NonNull
    private val messenger: BinaryMessenger
    private val channel: MethodChannel

    init {
        this.messenger = messenger
        this.channel = channel
    }

    @Override
    fun create(context: Context?, viewId: Int, args: Object?): PlatformView {
        return RelatedDigitalStoryView(context, viewId, args, channel)
    }
}